report_compassion/models/contract_group.py

```diff
@@ -152,32 +135,34 @@ def get_communication(self, start, stop, sponsorships):
             'subject': _("for") + " ",
             'date': '',
         }
-        with setlocale(self.partner_id.lang):
-            if start and stop and start == stop:
-                vals['date'] = date_start.strftime("%B %Y").decode('utf-8')
-            elif start and stop:
-                vals['date'] = date_start.strftime("%B %Y").decode('utf-8') + \
-                    " - " + date_stop.strftime("%B %Y").decode('utf-8')
-            if 'Permanent' in payment_mode.name:
-                vals['payment_type'] = _('ISR for standing order')
-                vals['date'] = ''
+        locale = self.partner_id.lang
```

Comment:
Same principle, define a local context for the `_()` translation function to work.
```suggestion
        locale = self.partner_id.lang
        context = {'lang': locale}
```